File: src/main/java/gg/mineral/practice/commands/stats/LeaderboardsCommand.java
package gg.mineral.practice.commands.stats;

import gg.mineral.practice.commands.PlayerCommand;
import gg.mineral.practice.inventory.menus.LeaderboardMenu;
import gg.mineral.practice.managers.ProfileManager;
import gg.mineral.practice.managers.QueuetypeManager;
import gg.mineral.practice.queue.Queuetype;

public class LeaderboardsCommand extends PlayerCommand {

	public LeaderboardsCommand() {
		super("leaderboards");
		setAliases("leaderboard", "lb");
	}

	@Override
	public void execute(org.bukkit.entity.Player pl, String[] args) {
		QueuetypeManager.getQueuetypes().values().stream().filter(Queuetype::isRanked).findAny()
				.ifPresent(queuetype -> ProfileManager.getOrCreateProfile(pl).openMenu(new LeaderboardMenu(queuetype)));
	}
}

File: src/main/kotlin/gg/mineral/practice/commands/stats/LeaderboardsCommand.kt
package gg.mineral.practice.commands.stats

import dev.rollczi.litecommands.annotations.command.Command
import dev.rollczi.litecommands.annotations.context.Context
import dev.rollczi.litecommands.annotations.execute.Execute
import gg.mineral.practice.entity.Profile
import gg.mineral.practice.inventory.menus.LeaderboardMenu
import gg.mineral.practice.managers.QueuetypeManager.queuetypes

@Command(name = "leaderboards", aliases = ["leaderboard", "lb"])
class LeaderboardsCommand {
    @Execute
    fun execute(@Context profile: Profile) =
        queuetypes.values.first { it.ranked }.let { queuetype -> profile.openMenu(LeaderboardMenu(queuetype)) }
}
